cvat/apps/engine/serializers.py

```diff
@@ -1199,6 +1190,47 @@ def _iterate_chunk_frames():
 
         return instance
 
+    @staticmethod
+    def _write_updated_static_chunk(
+        db_task_id: int,
+        db_segment_id: int,
+        chunk_id: int,
+        chunk_frames: list[int],
+        quality: FrameQuality,
+        frame_path_map: dict[int, str],
+        segment_frame_map: dict[int,int]
+    ):
+        from cvat.apps.engine.frame_provider import prepare_chunk
+
+        db_task = models.Task.objects.get(pk=db_task_id)
+        task_frame_provider = TaskFrameProvider(db_task)
+        db_data = db_task.data
+
+        def _iterate_chunk_frames():
+            for chunk_frame in chunk_frames:
+                db_frame_path = frame_path_map[chunk_frame]
+                chunk_real_frame = segment_frame_map.get(chunk_frame, chunk_frame)
+                yield (
+                    task_frame_provider.get_frame(
+                        chunk_real_frame, quality=quality
+                    ).data,
```

Comment:
If there are 2 server threads, it's possible that one of the threads / workers will update some of the chunks, while the other thread is working on other chunks. In this case the lines 1206 and 1214 can return unexpected results.